### PR TITLE
Fix PCSX core for big-endian archs

### PR DESCRIPTION
**Pull request by pcercuei**
A couple of fixes to the PCSX core to make it able to run on big-endian systems.
Tested on 32-bit PPC.

The various GPU plugins would also need to be patched in order to access the VRAM properly.

### CODE REVIEW

**Comment by pcercuei** [comment]
@notaz updated. There might still be some endianness bugs here and there, but games boot on PowerPC and are playable.

**Comment by notaz** [comment]
The CI build failure here looks legit: https://github.com/notaz/pcsx_rearmed/runs/6630931145?check_suite_focus=true

It would be nice if you did not break my hacked up arm* CI too, only set SDL_CONFIG if it's not already set in the environment and pass it pass it to configure from .github/workflows/ci-linux-*.yml , or I can patch it up later myself.

File: plugins/gpulib/vout_pl.c [hunk 1]
```diff
@@ -82,7 +83,7 @@ void vout_update(void)
 
   vram += y * 1024 + x;
 
-  cbs->pl_vout_flip(vram, 1024, gpu.status.rgb24, w, h);
+  cbs->pl_vout_flip(vram, 1024, gpu.status & PSX_GPU_STATUS_RGB24, w, h);
```
Comment (notaz) [inline plugins/gpulib/vout_pl.c:86]: can you make this `!!(gpu.status & PSX_GPU_STATUS_RGB24)` so it stays 0/1?

File: plugins/gpulib/vout_pl.c [hunk 2]
```diff
@@ -95,7 +96,7 @@ void vout_blank(void)
     w *= 2;
     h *= 2;
   }
-  cbs->pl_vout_flip(NULL, 1024, gpu.status.rgb24, w, h);
+  cbs->pl_vout_flip(NULL, 1024, gpu.status & PSX_GPU_STATUS_RGB24, w, h);
```
Comment (notaz) [inline plugins/gpulib/vout_pl.c:99]: also here

**Comment by pcercuei** [comment]
Sure, I'll update this PR later today.

**Comment by pcercuei** [comment]
Updated.

**Comment by joolswills** [comment]
Just to note - this was missing a change for armv6 in gpu_unai/gpulib_if.cpp - I have added a fix here https://github.com/notaz/pcsx_rearmed/pull/277

**Pull request merged**